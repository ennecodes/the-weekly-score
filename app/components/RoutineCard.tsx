import { TaskSwitch } from "./TaskSwitch";

export interface Routine {
  id: string;
  name: string;
  tasks: Task[];
}

export interface Task {
    id: string;
    name: string;
}

export const RoutineCard = ({ routine }: { routine: Routine }) => {
  return (
    <div className="rounded-lg bg-pink-100 p-4 shadow-lg">
      <h2 className="mb-4 text-center text-xl font-bold">{routine.name}</h2>
      <ul>
        {routine.tasks.map((task, index) => (
          <li key={index} className="flex items-center justify-between">
            <TaskSwitch task={task} />
          </li>
        ))}
      </ul>
    </div>
  );
};
