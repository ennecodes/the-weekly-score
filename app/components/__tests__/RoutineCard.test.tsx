import { render, screen, fireEvent } from "@testing-library/react";
import { describe, expect } from "vitest";

import { RoutineCard } from "../RoutineCard";

const testRoutine = {
    id: "1",

    name: "Test Routine",
    tasks: [{id: "1", name: "Task 1"}, {id: "2", name: "Task 2"}, {id: "3", name: "Task 3"}, {id: "4", name: "Task 4"},],
  };

describe("RoutineCard", () => {
  it("renders correctly", () => {
    const { container } = render(
      <RoutineCard
        routine={testRoutine}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it("displays the routine heading", () => {
    render(
      <RoutineCard
        routine={testRoutine}
      />,
    );
    expect(screen.getByText("Test Routine")).toBeInTheDocument();
  });
  it("displays all the tasks for the routine", () => {
    render(
        <RoutineCard
        routine={testRoutine}
      />,
    );
    expect(screen.getByText("Task 1")).toBeInTheDocument();
    expect(screen.getByText("Task 2")).toBeInTheDocument();
    expect(screen.getByText("Task 3")).toBeInTheDocument();
    expect(screen.getByText("Task 4")).toBeInTheDocument();
  });
  it("allows the user to mark tasks as complete", () => {
    render(
        <RoutineCard
        routine={testRoutine}
      />,
    );
    const task1Switch = screen.getByTestId("task-1-switch");
    fireEvent.click(task1Switch);
    expect(task1Switch).toBeChecked();
  });
});
