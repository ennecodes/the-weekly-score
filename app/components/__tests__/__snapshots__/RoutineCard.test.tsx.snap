// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoutineCard > renders correctly 1`] = `
<div>
  <div
    class="rounded-lg bg-pink-100 p-4 shadow-lg"
  >
    <h2
      class="mb-4 text-center text-xl font-bold"
    >
      Test Routine
    </h2>
    <ul>
      <li
        class="flex items-center justify-between"
      >
        <span>
          Task 1
        </span>
        <button
          aria-checked="false"
          class="group inline-flex h-6 w-11 items-center rounded-full bg-gray-200 transition data-[checked]:bg-pink-600"
          data-headlessui-state=""
          data-testid="task-1-switch"
          id="headlessui-switch-:r0:"
          role="switch"
          tabindex="0"
          type="button"
        >
          <span
            class="size-4 translate-x-1 rounded-full bg-white transition group-data-[checked]:translate-x-6"
          />
        </button>
      </li>
      <li
        class="flex items-center justify-between"
      >
        <span>
          Task 2
        </span>
        <button
          aria-checked="false"
          class="group inline-flex h-6 w-11 items-center rounded-full bg-gray-200 transition data-[checked]:bg-pink-600"
          data-headlessui-state=""
          data-testid="task-2-switch"
          id="headlessui-switch-:r1:"
          role="switch"
          tabindex="0"
          type="button"
        >
          <span
            class="size-4 translate-x-1 rounded-full bg-white transition group-data-[checked]:translate-x-6"
          />
        </button>
      </li>
      <li
        class="flex items-center justify-between"
      >
        <span>
          Task 3
        </span>
        <button
          aria-checked="false"
          class="group inline-flex h-6 w-11 items-center rounded-full bg-gray-200 transition data-[checked]:bg-pink-600"
          data-headlessui-state=""
          data-testid="task-3-switch"
          id="headlessui-switch-:r2:"
          role="switch"
          tabindex="0"
          type="button"
        >
          <span
            class="size-4 translate-x-1 rounded-full bg-white transition group-data-[checked]:translate-x-6"
          />
        </button>
      </li>
      <li
        class="flex items-center justify-between"
      >
        <span>
          Task 4
        </span>
        <button
          aria-checked="false"
          class="group inline-flex h-6 w-11 items-center rounded-full bg-gray-200 transition data-[checked]:bg-pink-600"
          data-headlessui-state=""
          data-testid="task-4-switch"
          id="headlessui-switch-:r3:"
          role="switch"
          tabindex="0"
          type="button"
        >
          <span
            class="size-4 translate-x-1 rounded-full bg-white transition group-data-[checked]:translate-x-6"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;
